Fix STL loader

Martin-Idel opened the pull request:
Probably fix #355 

The STL loader has a bug where it loads the meshes into the wrong resource group. This results in a crash when a mesh with the same name (the resource location) is loaded twice at runtime. Normally, the already loaded mesh should be returned.

The first commit adds a test demonstrating this bug, which is probably the root cause for #355.
The second commit fixes that bug. 

I don't have a good way to deal with the fact that we define the "ROS_PACKAGE_NAME" in several files. We could define it via CMake but that's not very nice either so for now I suggest duplicating the definition (as is already done throughout `rviz_rendering`). Since we have a test now, any problematic updates should be caught.

Martin-Idel committed:
Add test to show bug

Martin-Idel committed:
Fix bug with loading an stl mesh twice

The mesh loader checks whether the mesh has already been loaded.
However, it only checks in the ResourceGroup `rviz_rendering`, but
the STL loader loads into the DefaultResourceGroup by accident.

wjwwood commented:
Thanks for taking a crack at it @Martin-Idel, we really appreciate the contribution :)

Here's some CI:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=7306)](http://ci.ros2.org/job/ci_linux/7306/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=3480)](http://ci.ros2.org/job/ci_linux-aarch64/3480/)
* macOS [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_osx&build=6004)](https://ci.ros2.org/job/ci_osx/6004/) (restarted as it failed due to an agent going offline)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=7126)](http://ci.ros2.org/job/ci_windows/7126/)

mikaelarguedas commented:
Awesome thanks @Martin-Idel !!

I tried it locally and it fixes the issue!

There are multiple error messages (~100) being printed to console when (re)loading the resource:
```
[ERROR] [rviz2]: Can't assign material 'BaseWhiteNoLighting' to SubEntity of 'mesh_resource_marker_0' because this Material does not exist in group 'rviz_rendering'. Have you forgotten to define it in a .material script?
```

wjwwood commented:
@mikaelarguedas I think I've seen those before, any idea if they are a side effect of this fix or if they were preexisting?

I vaguely remember that this could occur when we were statically linking ogre and therefore we could get different instances of the mesh resources (since it is stored statically in the ogre library), but I don't think that's happening any longer IIRC, so I don't know what the root cause is.

@Martin-Idel do you think we should investigate that error message before merging this or merge this and investigate it separately?

mikaelarguedas commented:
> any idea if they are a side effect of this fix or if they were preexisting?

Using the dashing packages from `ros2-testing` I don't see any error messages so I believe they are related to this patch

Martin-Idel commented:
@wjwwood Sure, I'll have a look.
It'll be related to the same problem that the resource manager has some problems...

Martin-Idel committed:
Add BaseWhiteNoLighting to rviz_rendering resource group

Previous to Ogre 1.11, materials would be searched in all groups
With 1.11, this is no longer true.

In RViz, we try to put materials in our own resource group, but we
need the Ogre fallback material.

Martin-Idel commented:
@mikaelarguedas I can't really reproduce the errors because I don't have an stl mesh which would use the BaseWhiteNoLighting fallback material, however I think I know what's going on, but I'd need verification that this fixes the issue.

Here is my understanding:

The "BaseWhiteNoLighting" material is the basic Ogre fallback material present in the default resource group.
My first change puts stl meshes into the `rviz_rendering` resource group as all other meshes. Since Ogre 1.11 materials are only searched for in the `rviz_rendering` resource group hence the "BaseWhiteNoLighting" material cannot be found.

The latest change adds the material also to the `rviz_rendering` resource group so I hope the failures go away.

mikaelarguedas commented:
:100: :tada: Tried this with the [faulty mesh](https://github.com/Karsten1987/roscon2018/blob/master/confbot_description/meshes/Shark_t.stl) and with the latest patch it now unloads and reloads successfully without printing any error message.

Thanks @Martin-Idel !

wjwwood commented:
Rerunning Linux CI:

[![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_linux&build=7333)](https://ci.ros2.org/job/ci_linux/7333/)
